apps/src/clientApi.js

```diff
@@ -171,20 +183,28 @@ class AssetsApi extends CollectionsApi {
       },
       error);
   }
+
+  /*
+   * Create or update an asset
+   * @param filename {String} filename to be created or updated
+   * @param data {Blob} asset data
+   * @param success {Function} callback when successful (includes xhr parameter)
+   * @param error {Function} callback when failed
+   */
+  putAsset(filename, data, success, error) {
+    this._withBeforeFirstWriteHook(() => {
+      ajaxInternal('PUT', this.basePath(filename), xhr => {
+          if (success) {
+            success(xhr);
+          }
+        },
+        error,
+        data);
```

Comment:
Nit: The formatting here is throwing me off, but I'm not sure it matters enough to change.  It did occur to me that this would make a clean one-liner if we provided default callbacks:
```js
putAsset(filename, data, success = () => {}, error = () => {}) {
  this._withBeforeFirstWriteHook(() => {
    ajaxInternal('PUT', this.basePath(filename), success, error, data);
  });
}
```